lib/sda/service.go

```diff
@@ -278,3 +282,120 @@ func CreateServiceMessage(service string, r interface{}) (*ServiceMessage, error
 	}, nil
 
 }
+
+/*
+A simple client structure to be used when wanting to connect to services. It
+holds the private and public key and allows to connect to a service through
+the network.
+The error-handling is done using the ErrorRet structure which can be returned
+in place of the standard reply. The Client.Send method will catch that and return
+ the appropriate error.
+*/
+
+// Client for a service
+type Client struct {
+	Private abstract.Secret
+	*network.Entity
+	ServiceID ServiceID
+}
+
+// NewClient returns a random client using the service s
+func NewClient(s string) *Client {
+	kp := config.NewKeyPair(network.Suite)
+	return &Client{
+		Entity:    network.NewEntity(kp.Public, ""),
+		Private:   kp.Secret,
+		ServiceID: ServiceFactory.ServiceID(s),
+	}
+}
+
+// NetworkSend opens the connection to 'dst' and sends the message 'req'. The
+// reply is returned, or an error if the timeout of 10 seconds is reached.
+func (c *Client) Send(dst *network.Entity, msg network.ProtocolMessage) (*network.Message, error) {
+	client := network.NewSecureTCPHost(c.Private, c.Entity)
```

Comment:
Why waiting on the `send` ? Why not directly into the `NewClient` ? We may want to use a client many times and not only one. Creating a longterm  `TCPHost` seems more efficient.
Do we want to use one `Client` for different `Services` addresses ? Because, directly connecting to the `Services` at the beginning also seems more efficient, but I don't know what our use cases are. For me, it seems logical that one `Client` connects to one `Service` but then again, that's my opinion. What do you think ?